packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -239,4 +246,10 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
     });
     forceUpdate();
   }, [apiRef, setGridState, forceUpdate, isRowSelectable]);
+
+  React.useEffect(() => {
+    if (totalSelectedRows > visibleRowCount) {
+      selectRows(visibleRowIds, true, true);
+    }
+  }, [totalSelectedRows, visibleRowIds, selectRows, visibleRowCount]);
```

Comment:
Really not sure about this useEffect. 
If any of the dependency changes then it will select the rows.

Reply:
better to use apiRef for the api call as it avoids adding the selectRows as a dependency

Reply:
> better to use apiRef for the api call as it avoids adding the selectRows as a dependency

Then apiRef will be a dependency.

Reply:
yes but that's fine as it's a ref it doesn't change like the function itself

Reply:
Okay. Added `apiRef`. I was wondering if refs don't change why we need it as a dependency and why does hooks eslint complains. Usually we don't add refs as dependency.